feat: MCP L1

tynes opened the pull request:
<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

**Description**

MCP L1 contracts release

0xfuturistic committed:
contracts-bedrock: move immutables to storage in OptimismPortal

0xfuturistic committed:
contracts-bedrock: move immutables to storage in StandardBridge

0xfuturistic committed:
contracts-bedrock: update L1StandardBridge to support changes to StandardBridge

0xfuturistic committed:
contracts-bedrock: update L2StandardBridge to support changes to StandardBridge

0xfuturistic committed:
contracts-bedrock: move immutables to storage in ERC721Bridge

0xfuturistic committed:
contracts-bedrock: update L1ERC721Bridge to support changes to ERC721Bridge

0xfuturistic committed:
contracts-bedrock: update L2ERC721Bridge to support changes to ERC721Bridge

0xfuturistic committed:
contracts-bedrock: update Deploy script to support changes to OptimismPortal

OptimismPortal had its immutables moved to storage

0xfuturistic committed:
contracts-bedrock: update tests to support changes to OptimismPortal

OptimismPortal had its immutables moved to storage

0xfuturistic committed:
contracts-bedrock: update Deploy script to support changes to StandardBridge, L1StandardBridge, L2StandardBridge

StandardBridge, L1StandardBridge, L2StandardBridge had its immutables moved to storage

0xfuturistic committed:
contracts-bedrock: add TODO comment to L1StandardBridge

0xfuturistic committed:
contracts-bedrock: update tests to support changes to StandardBridge

StandardBridge had its immutables moved to storage

0xfuturistic committed:
contracts-bedrock: update tests to support changes to L1StandardBridge

L1StandardBridge had its immutables moved to storage

0xfuturistic committed:
contracts-bedrock: update Deploy script to support changes to L1ERC721Bridge

L1ERC721Bridge had its immutables moved to storage

0xfuturistic committed:
contracts-bedrock: fix Deploy script's function for OptimismPortal

Since values are set at initialization and not at deployment, ChainAssertions.checkOptimismPortal reverts if called at this point with the dependent contracts. Instead, this call should be done immediately after initialization (as is already done in the Deploy script).

0xfuturistic committed:
contracts-bedrock: remove requirements in __ERC721Bridge_init

This is needed because a call to __ERC721Bridge_init is triggered by the constructors of L1ERC721Bridge and L2ERC721Bridge. However, since the variables required to be non-null by __ERC721Bridge_init are set to null values in those constructors, this will call revert and the set-ups of L1ERC721Bridge and L2ERC721Bridge will fail. This commit fixes that issue.

0xfuturistic committed:
contracts-bedrock: fix Deploy script for L1ERC721Bridge

Since values are set at initialization and not at deployment, ChainAssertions.checkL1ERC721Bridge reverts if called at this point with the dependent contracts. Instead, this call should be done immediately after initialization (as is already done in the Deploy script).

0xfuturistic committed:
contracts-bedrock: fix Deploy script for L1StandardBridge

Since values are set at initialization and not at deployment, ChainAssertions.checkL1StandardBridge reverts if called at this point with the dependent contracts. Instead, this call should be done immediately after initialization (as is already done in the Deploy script). Additional bug fixes are included, including the removal of unnecessary require checks originally obtained from the legacy PR (#6573)

0xfuturistic committed:
contracts-bedrock: fix test for L1StandardBridge

test_initialize_fix_succeeds is removed, which was originally obtained (out of context) from the legacy PR (#6573)

0xfuturistic committed:
op-chain-ops: update upgrade script for OptimismPortal

OptimismPortal had its immutable variables moved to storage

0xfuturistic committed:
op-chain-ops: update upgrade script for L1ERC721Bridge

L1ERC721Bridge had its immutable variables moved to storage

0xfuturistic committed:
op-chain-ops: update upgrade script for L1StandardBridge

L1StandardBridge had its immutable variables moved to storage

0xfuturistic committed:
contracts-bedrock: fix OptimismPortal spec in Authorization.t.sol

Addresses the fact that OptimismPortal's spec for initialize should include the parameters for the variables moved from immutables into storage

0xfuturistic committed:
contracts-bedrock: fix L1StandardBridge spec in Authorization.t.sol

Addresses the fact that L1StandardBridge spec for initialize should include the parameters for the variables moved from immutables into storage

0xfuturistic committed:
contracts-bedrock: fix L1ERC721Bridge spec in Authorization.t.sol

Addresses the fact that L1ERC721Bridge spec for initialize should include the parameters for the variables moved from immutables into storage

0xfuturistic committed:
contracts-bedrock: remove requires from __StandardBridge_init in StandardBridge

necessary for defining messenger address as address(0) in constructor of L1StandardBridge, for example

0xfuturistic committed:
contracts-bedrock: make L2StandardBridge backwards compatible with immutable L2 deploy script

This is done by reintroducing a parameter for the address of the other bridge in the constructor of L2StandardBridge

0xfuturistic committed:
contracts-bedrock: update Initializable.t.sol with parameters for L2StandardBridge

0xfuturistic committed:
contracts-bedrock: update semver for L1ERC721Bridge, L2ERC721Bridge

0xfuturistic committed:
contracts-bedrock: bump semver for OptimismPortal, improve comments

0xfuturistic committed:
op-chain-ops: introduce storage values for L2StandardBridge

corresponding to the immutable variables moved into storage in the contract

0xfuturistic committed:
op-chain-ops: redefine L2ERC721Bridge immutable config

immutable variables were removed

0xfuturistic committed:
op-chain-ops: introduce storage values for L2ERC721Bridge

corresponding to the immutable variables moved into storage in the contract

0xfuturistic committed:
contracts-bedrock: set L2OutputOracle upgradeToAndCall initialize version back to 2 in tests

0xfuturistic committed:
contracts-bedrock: extend constructor test for OptimismPortal

0xfuturistic committed:
contracts-bedrock: set initialize version back to 2 in OptimismPortal's test_upgradeToAndCall_upgrading_succeeds

0xfuturistic committed:
contracts-bedrock: add more requirements in OptimismPortal deploy script

0xfuturistic committed:
contracts-bedrock: represent contracts with addresses in initializeOptimismPortal of deploy script

0xfuturistic committed:
contracts-bedrock: improve labeling of variables in l1.go for OptimismPortal

0xfuturistic committed:
op-chain-ops: relabel variables in L1ERC721Bridge upgrade script

0xfuturistic committed:
contracts-bedrock: fix annotation in ERC721Bridge

0xfuturistic committed:
contracts-bedrock: reintroduce ChainAssertions for L1ERC721Bridge at deploy time

0xfuturistic committed:
contracts-bedrock: reintroduce parameter to L2ERC721Bridge constructor

this makes the changes backwards compatible with existing deploy scripts

0xfuturistic committed:
contracts-bedrock: reintroduce ChainAssertions for deployment of OptimismPortal

this is possible thanks to introducing a flag for ChainAssertions to know whether the contract in question has been already initialized or not

0xfuturistic committed:
op-bindings: make OptimismPortal bindings

0xfuturistic committed:
contracts-bedrock: update semver-lock for changes to OptimismPortal

0xfuturistic committed:
contracts-bedrock: update slither report

0xfuturistic committed:
op-chain-ops: fix bug, missing parameter for L2ERC721Bridge deploy

0xfuturistic committed:
contracts-bedrock: make snapshots for L1ERC721Bridge, L2ERC721Bridge

0xfuturistic committed:
op-bindings: make OptimismPortal

abigen version was changed to 1.10.16-stable

0xfuturistic committed:
contracts-bedrock: make snapshots

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: improve comments for StandardBridge

0xfuturistic committed:
contracts-bedrock: bump semver for L2StandardBridge, improve comments

0xfuturistic committed:
contracts-bedrock: bump semver of L1StandardBridge, improve comments

additionally, the messenger's address is set to address(0) at initialization

0xfuturistic committed:
contracts-bedrock: reintroduce ChainAssertions for L1StandardBridge at deploy

0xfuturistic committed:
op-chain-ops: improve labeling of vars in upgrade script for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: move repeated code in ChainAssertions's checkL1ERC721Bridge

repeated code is moved outside the if-else block

0xfuturistic committed:
op-bindings: make for StandardBridge, L1StandardBridge, L2StandardBridge

0xfuturistic committed:
contracts-bedrock: make snapshots for L1StandardBridge, L2StandardBridge

0xfuturistic committed:
op-bindings: make for L1StandardBridge, L2StandardBridge

0xfuturistic committed:
contracts-bedrock: add test for constructor of OptimismPortal

0xfuturistic committed:
contracts-bedrock: improve dev comments in OptimismPortal tests

0xfuturistic committed:
contracts-bedrock: drop isInitialized in ChainAssertions for OptimismPortal

now isProxy is used directly

0xfuturistic committed:
contracts-bedrock: minor improvement in test for OptimismPortal

0xfuturistic committed:
contracts-bedrock: tests for OptimismPortal updated to use non-legacy getters

0xfuturistic committed:
contracts-bedrock: leverage impl in constructor test for OptimismPortal

0xfuturistic committed:
contracts-bedrock: update semver for OptimismPortal

additionally improve documentation in file

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: update slither report

0xfuturistic committed:
op-bindings: make for OptimismPortal

0xfuturistic committed:
op-chain-ops: fix otherBridge in L1StandardBridge upgrade script

0xfuturistic committed:
contracts-bedrock: replace isInitialized by isProxy in ChainAssertions for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: add constructor test for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: extend initialize test for L2StandardBridge

0xfuturistic committed:
contracts-bedrock: replace isInitialized by isProxy in ChainAssertions for L1ERC721Bridge

0xfuturistic committed:
op-chain-ops: fix bug in L1ERC721Bridge upgrade script

0xfuturistic committed:
contracts-bedrock: update gap in ERC721Bridge

done to maintain existing storage layout

0xfuturistic committed:
contracts-bedrock: update snapshots for L1ERC721Bridge, L2ERC721Bridge

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-bindings: make for L1ERC721Bridge, L2ERC721Bridge

0xfuturistic committed:
contracts-bedrock: improve documentation in OptimismPortal

0xfuturistic committed:
contracts-bedrock: update slither report

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: improve documentation for CrossDomainMessenger, L1CrossDomainMessenger, L2CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: improve documentation for StandardBridge

0xfuturistic committed:
contracts-bedrock: shorten gap in StandardBridge

to preserve existing storage layout

0xfuturistic committed:
contracts-bedrock: update snapshots for L1StandardBridge, L2StandardBridge

0xfuturistic committed:
op-bindings: make for StandardBridge, L1StandardBridge, L2StandardBridge

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: introduce addresses of L1 system contracts

0xfuturistic committed:
contracts-bedrock: update Deploy script for changes to SystemConfig

0xfuturistic committed:
contracts-bedrock: update Initializable test for changes to SystemConfig

0xfuturistic committed:
contracts-bedrock: update invariant test for changes to SystemConfig

0xfuturistic committed:
contracts-bedrock: update test_initialize_values_succeeds for SystemConfig test

0xfuturistic committed:
contracts-bedrock: add new tests for SystemConfig

0xfuturistic committed:
contracts-bedrock: update ChainAssertions for SystemConfig

0xfuturistic committed:
contracts-bedrock: add salt to deployment of SystemConfig

0xfuturistic committed:
contracts-bedrock: update Specs.t.sol with new methods on SystemConfig

0xfuturistic committed:
op-bindings: make for SystemConfig

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: update invariant docs

0xfuturistic committed:
contracts-bedrock: improve support for startBlock in systemConfig

0xfuturistic committed:
contracts-bedrock: fix comments in CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: change type of legacy getter for L1CrossDomainMessenger PORTAL

0xfuturistic committed:
contracts-bedrock: add tests for L1CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: add tests for L2CrossDomainMessenger

0xfuturistic committed:
op-bindings: make for L1CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-chain-ops: reorder storage values

maintain consistency with previous order

0xfuturistic committed:
contracts-bedrock: add tests for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: add tests for L2StandardBridge

0xfuturistic committed:
contracts-bedrock: change !isProxy isProxy order in ChainAssertions for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: change !isProxy isProxy order in ChainAssertions for L1ERC721Bridge

0xfuturistic committed:
contracts-bedrock: add tests for L1ERC721Bridge

0xfuturistic committed:
contracts-bedrock: add tests for L2ERC721Bridge

0xfuturistic committed:
contracts-bedrock: fix test for L1ERC721Bridge

0xfuturistic committed:
op-chain-ops: reorder setting of storage variables in config

makes the order consistent with previous order for immutables

0xfuturistic committed:
contracts-bedrock: use address from predeploys in Deploy script for L1ERC721Bridge

0xfuturistic committed:
contracts-bedrock: reorder isProxy in ChainAssertions for OptimismPortal

0xfuturistic committed:
contracts-bedrock: reorder isProxy in ChainAssertions for L1CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: use predeploys in Deploy script for L1CrossDomainMessenger

0xfuturistic committed:
op-bindings: make for L1CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: use predeploys in Deploy script for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: add require and tests for ERC721Bridge

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-bindings: make for L1ERC721Bridge, L2ERC721Bridge

0xfuturistic committed:
contracts-bedrock: reorder isProxy in ChainAssertions for SystemConfig

0xfuturistic committed:
contracts-bedrock: set otherBridge to null address in constructor of L1ERC721Bridge

0xfuturistic committed:
contracts-bedrock: hardcode messenger in L2ERC721Bridge to Predeploys

0xfuturistic committed:
contracts-bedrock: drop require for bridge to be null in ERC721Bridge

0xfuturistic committed:
op-bindings: make for L1ERC721Bridge, L2ERC721Bridge

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: use predeploys to obtain otherBridge in L1StandardBridge

0xfuturistic committed:
contracts-bedrock: use predeploys to obtain otherBridge in L2StandardBridge

0xfuturistic committed:
op-bindings: make for L1StandardBridge, L2StandardBridge

0xfuturistic committed:
contracts-bedrock: drop redundant test (already in Initialize)

L1 proxy is already tested for in intiailize, but L1 impl not

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: drop unnecessary parameter in L1CrossDomainMessenger initialize

0xfuturistic committed:
contracts-bedrock: change type of otherMessenger in CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: drop constructor arg for L1CrossDomainMessenger, L2CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: change type of otherBridge in ERC721Bridge

0xfuturistic committed:
contracts-bedrock: drop constructor arg, use Predeploy in L1ERC721Bridge

0xfuturistic committed:
contracts-bedrock: replace dead address by DEFAULT_L2_SENDER from constants in tests

0xfuturistic committed:
contracts-bedrock: drop redundant test for OptimismPortal

already covered in Initialize.t.sol

0xfuturistic committed:
contracts-bedrock: update Deployer's loadInitializedSlot

if contract is a proxy, it now gets the implementation address using either EIP1967Helper or AddressManager (for ResolvedDelegateProxy)

0xfuturistic committed:
contracts-bedrock: define constants in EIP1967Helper and drop Constants import

this allows us to not have to update the version of Arithmetic, ResourceMetering, and Burn, which would otherwise clash with Deployer.sol

0xfuturistic committed:
contracts-bedrock: use interface for AddressManager instead of updating version directly

0xfuturistic committed:
contracts-bedrock: update slither report

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-chain-ops: check fetched addresses from OptimismPortal against DeployConfig

0xfuturistic committed:
contracts-bedrock: drop redundant test for OptimismPortal

this is covered in a different PR refactoring Initializable.t.sol

0xfuturistic committed:
contracts-bedrock: fix test for L1CrossDomainMessenger

0xfuturistic committed:
op-bindings: make for CrossdomainMessenger, L1CrossdomainMessenger, L2CrossdomainMessenger

0xfuturistic committed:
op-chain-ops: check fetched addresses from L1CrossDomainMessenger against DeployConfig

0xfuturistic committed:
contracts-bedrock: drop redundant tests for L1CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: drop redundant test for L2CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: fix documentation for L2CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: update slither report

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-chain-ops: check fetched addresses from L1StandardBridge against DeployConfig

0xfuturistic committed:
contracts-bedrock: drop redundant test for L1StandardBridge

0xfuturistic committed:
contracts-bedrock: drop redundant test for L2StandardBridge

0xfuturistic committed:
contracts-bedrock: kwargs syntax update in StandardBridge

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
contracts-bedrock: fix Initializable.t.sol

0xfuturistic committed:
contracts-bedrock: fix rebase

0xfuturistic committed:
contracts-bedrock: drop redundant tests

0xfuturistic committed:
contracts-bedrock: fix rebase

0xfuturistic committed:
op-chain-ops: check fetched addresses from OptimismPortal against DeployConfig

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: make snapshots

0xfuturistic committed:
op-chain-ops: fix after rebase

0xfuturistic committed:
contracts-bedrock: restore empty test contract for L1StandardBridge

0xfuturistic committed:
op-chain-ops: check fetched addresses from SystemConfig against DeployConfig

0xfuturistic committed:
contracts-bedrock: update default startBlock number for SystemConfig

0xfuturistic committed:
contracts-bedrock: update tests for SystemConfig

0xfuturistic committed:
op-chain-ops: fix l1 initialize for L1StandardBridge

0xfuturistic committed:
op-chain-ops: improve comparison over a non-null config for SystemConfig

0xfuturistic committed:
contracts-bedrock: drop redundant line in OptimismPortal

0xfuturistic committed:
op-chain-ops: drop unnecessary parameter for l1ERC721BridgeABI in L1 upgrades

0xfuturistic committed:
op-chain-ops: move check out of restrictive clause in L1 upgrade for L1ERC721Bridge

0xfuturistic committed:
op-chain-ops: drop unnecessary parameter in L1CrossDomainMessenger L1 upgrade

0xfuturistic committed:
contracts-bedrock: move check out of config existence clause in L1 upgrade script for L1CrossDomainMessenger

0xfuturistic committed:
op-chain-ops: move check out of clause for L1 upgrade path L1StandardBridge

tynes committed:
contracts-bedrock: update semver-lock

tynes committed:
contracts-bedrock: fix constructor

0xfuturistic committed:
contracts-bedrock: set start block in constructor and not initialize for SystemConfig

0xfuturistic committed:
contracts-bedrock: update slither-report

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
contracts-bedrock: drop parameter from SystemConfig initialize

0xfuturistic committed:
contracts-bedrock: add new tests for SystemConfig startBlock

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: update invariant docs

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
contracts-bedrock: update slither report

0xfuturistic committed:
op-chain-ops: fix immutables

0xfuturistic committed:
op-chain-ops: fix rebase wrt immutables of L2StandardBridge

0xfuturistic committed:
op-chain-ops: fix by dropping duplicate import in L1 upgrade

0xfuturistic committed:
op-chain-ops: drop unnecessary import in l1 upgrade

0xfuturistic committed:
contracts-bedrock: fix SystemConfig tests

0xfuturistic committed:
contracts-bedrock: fix after rebase Initializable

0xfuturistic committed:
contracts-bedrock: use START_BLOCK_SLOT for startBlock var in SystemConfig

0xfuturistic committed:
contracts-bedrock: fix test

0xfuturistic committed:
op-chain-ops: fix op-upgrade

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
op-chain-ops: fix SystemConfig upgrade

0xfuturistic committed:
contracts-bedrock: drop public StartBlock setter function in SystemConfig

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
op-chain-ops: lint

mds1 commented on packages/contracts-bedrock/snapshots/abi/L1CrossDomainMessenger.json:
Just noting that I've reviewed the ABI diff and don't see any breaking or backwards-incompatible changes there. All ABI changes are either:
1. `constructor` signature changes
2. `initialize` signature changes
3. New methods and events

Which should not affect users.

Planning to check the storage layout next

0xfuturistic committed:
contracts-bedrock: update OptimismPortal version

0xfuturistic committed:
contracts-bedrock: improve ChainAssertions for OptimismPortal

0xfuturistic committed:
contracts-bedrock: reorder variable definition in L1CrossDomainMessenger

this is so the definition of the variable ´version´ happens after all other variables have been defined (more consistent with general pattern)

0xfuturistic committed:
contracts-bedrock: bump semver of L1CrossDomainMessenger

0xfuturistic committed:
contracts-bedrock: bump semver of L2CrossDomainMessenger

tynes commented on packages/contracts-bedrock/src/L1/SystemConfig.sol:
```diff
@@ -70,31 +107,33 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
+                optimismMintableERC20Factory: address(0)
+            })
         });
+        Storage.setUint(START_BLOCK_SLOT, type(uint256).max);
```
We should set the start block slot before calling initialize so that inside of initialize the setting of the start block is skipped. We should add a comment clarifying that its set particularly this way before initialize. Chains deployed before the introduction of the start block need to have the slot set manually and newly deployed chains will get the value set automatically

tynes commented on packages/contracts-bedrock/src/libraries/Arithmetic.sol:
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.0;
+pragma solidity 0.8.15;
```
why is this version changed? Ideally it can not be changed

tynes commented on packages/contracts-bedrock/test/L1/L2OutputOracle.t.sol:
```diff
@@ -28,64 +47,18 @@ contract L2OutputOracle_constructor_Test is CommonTest {
-
-    /// @dev Tests that the constructor reverts if the l2BlockTime is invalid.
-    function test_constructor_l2BlockTimeZero_reverts() external {
-        address proposer = deploy.cfg().l2OutputOracleProposer();
```
Are you sure that we have test coverage of these tests elsewhere?

tynes replied:
Oh i see, the tests where moved to the bottom of the file

0xfuturistic committed:
contracts-bedrock: improve test code for L1ERC721Bridge

0xfuturistic committed:
contracts-bedrock: restore Arithmetic.sol solc version

0xfuturistic committed:
contracts-bedrock: update START_BLOCK_SLOT in SystemConfig constructor

0xfuturistic committed:
contracts-bedrock: bump semver of SystemConfig

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-bindings: make

0xfuturistic committed:
contracts-bedrock: drop unnecessary import in L2ERC721Bridge.t.sol

mds1 commented on packages/contracts-bedrock/snapshots/storageLayout/L1ERC721Bridge.json:
```diff
@@ -14,11 +14,25 @@
   },
   {
-    "bytes": "1536",
-    "label": "__gap",
+    "bytes": "20",
+    "label": "messenger",
+    "offset": 2,
+    "slot": "0",
+    "type": "contract CrossDomainMessenger"
+  },
```
This is being packed into the intializer slot, are we ok with that? It's not a breaking change but I recall discussing if we wanted to avoid this. If we never drop into assembly to modify slot 0 (and I don't think we do) this is safe, so to be clear I'm ok keeping it

tynes replied:
We will need to clear the initialized slot during the 2 step upgrade so we will either need to:
- re-set the messenger in the `initialize` call
- avoid zeroing it out by packing the call to `setStorage` correctly
- putting a spacer in the contract

I am fine with any of these options

mds1 commented on packages/contracts-bedrock/snapshots/storageLayout/L2ERC721Bridge.json:
```diff
@@ -14,10 +14,24 @@
   },
   {
-    "bytes": "1536",
-    "label": "__gap",
+    "bytes": "20",
+    "label": "messenger",
+    "offset": 2,
+    "slot": "0",
+    "type": "contract CrossDomainMessenger"
+  },
```
Same comment re packing with initializer data

mds1 commented on packages/contracts-bedrock/snapshots/storageLayout/OptimismMintableERC20Factory.json:
```diff
@@ -1 +1,30 @@
+    "slot": "0",
+    "type": "bool"
+  },
+  {
+    "bytes": "20",
+    "label": "bridge",
+    "offset": 2,
+    "slot": "0",
+    "type": "address"
+  },
```
same initializer comment

mds1 reviewed (commented):
No storage layout issues. One question around packing with initializers. I have no problem with, and prefer it, but I recall briefly discussing it as an option for safety a while ago so figured I'd sanity check

0xfuturistic committed:
op-chain-ops: upgrades sets startBlock from config

tynes commented:
Looks like this needs a rebase + gas snapshot regeneration before we can merge it

0xfuturistic committed:
contracts-bedrock: add spacing to ERC721Bridge

0xfuturistic committed:
contracts-bedrock: add spacing to OptimismMintableERC20Factory

0xfuturistic committed:
contracts-bedrock: update snapshots

0xfuturistic committed:
contracts-bedrock: fix forge-std version following rebase

0xfuturistic committed:
contracts-bedrock: update gas-snapshot

0xfuturistic committed:
contracts-bedrock: update semver-lock

0xfuturistic committed:
op-bindings: make

0xfuturistic commented:
rebased with [develop](https://github.com/ethereum-optimism/optimism/tree/develop)

tynes commented on packages/contracts-bedrock/snapshots/storageLayout/L2ERC721Bridge.json:
```diff
@@ -14,10 +14,31 @@
-    "bytes": "1536",
-    "label": "__gap",
+    "bytes": "20",
+    "label": "spacer_0_2_20",
```
@mds1 do you think this should be a `bytes30`?

mds1 replied:
I agree with that. From re-reading [Storage Layout Management](https://www.notion.so/oplabs/Storage-Layout-Management-9fd6aa22b3cb4702b32fe65ae77ff21d?pvs=4#9b983922d0404316883968a09a3154dd) I don't think we ever came to consensus on a spacer format, but my suggestion would be to always use `bytesN` where `N` is the number of bytes needed to fill the rest of the slot without any gap. 

This is because if we later added a `bool` after this spacer, it would get packed into slot 0, and we're relying on the size of `messenger` to avoid packing.

mds1 commented on packages/contracts-bedrock/snapshots/storageLayout/L1ERC721Bridge.json:
```diff
@@ -14,11 +14,32 @@
+    "label": "spacer_0_2_20",
+    "offset": 2,
+    "slot": "0",
+    "type": "address"
```
Same comment on `bytes30` type here

mds1 commented on packages/contracts-bedrock/snapshots/storageLayout/OptimismMintableERC20Factory.json:
```diff
@@ -1 +1,37 @@
+    "label": "spacer_0_2_20",
+    "offset": 2,
+    "slot": "0",
+    "type": "address"
```
same `bytes30` suggestion again

0xfuturistic committed:
contracts-bedrock: change type of spacer_0_2_20 to bytes30

tynes committed:
contracts-bedrock: regenerate

tynes committed:
op-bindings: build

0xfuturistic committed:
contracts-bedrock: fix validate-spacers.ts

0xfuturistic committed:
contracts-bedrock: fix validate-spacers.ts

maurelian commented on packages/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol:
```diff
@@ -13,50 +13,58 @@ import { SuperchainConfig } from "src/L1/SuperchainConfig.sol";
-    /// @custom:semver 2.2.0
-    string public constant version = "2.2.0";
+    /// @custom:semver 2.3.0
+    string public constant version = "2.3.0";
```
@tynes I'm assuming that you discussed whether or not this is a change to the security model and decided no?

tynes replied:
We did talk about this, it should be fine to be a minor version bump everywhere

tynes committed:
op-chain-ops: extended pause op-upgrade

tynes commented:
Do not merge until https://github.com/ethereum-optimism/optimism/pull/9109 is merged first

maurelian reviewed (approved):
Approving on the basis of previous reviews on this branch.

protolambda reviewed (approved):
The Go changes look valid to me, but someone with more MCP L1 context should review the contract changes.

tynes commented:
> The Go changes look valid to me, but someone with more MCP L1 context should review the contract changes.

@maurelian @mds1 and @tynes have reviewed the contract changes

0xfuturistic committed:
indexer: pad deposit tx to fix out-of-gas error
